test/mocha/20-revocation.js

```diff
@@ -0,0 +1,367 @@
+    // exists on the host system
+    const bobRevocationZcap = {
+      '@context': bedrock.config.constants.SECURITY_CONTEXT_V2_URL,
+      // this is a unique ID
+      id: `urn:zcap:${uuid()}`,
+      invoker: bobKey.id,
+      // there is no root capability at the `invocationTarget` location,
+      // so this alternate URL is used that will automatically generate a
+      // root capability
+      parentCapability: `${aliceKeystoreAgent.keystore.id}/zcaps/revocations`,
+      allowedAction: 'write',
+      invocationTarget: `${aliceKeystoreAgent.keystore.id}/revocations`,
+    };
+
+    // Alice now signs the capability delegation that allows Bob to `sign`
+    // with her key.
+    const signer = aliceCapabilityAgent.getSigner();
+    const signedCapabilityFromAlice = await _delegate({
+      capabilityChain: [
+        // Alice's key is the root capability which is always the first
+        // item in the `capabilityChain`
+        aliceKey.kmsId
+      ],
+      signer,
+      zcap
+    });
+
+    // Alice now signs the capability delegation that allows Bob to `write`
+    // to Alice's keystore revocations endpoint
+    const signedBobRevocationZcap = await _delegate({
+      capabilityChain: [bobRevocationZcap.parentCapability],
+      signer,
+      zcap: bobRevocationZcap
+    });
+
+    // Bob now uses his delegated authority to sign a document with Alice's key
+    const bobSignedDocument = await _signWithDelegatedKey({
+      capability: signedCapabilityFromAlice,
+      invokeKey: bobKey,
+    });
+
+    bobSignedDocument.should.have.property('@context');
+    bobSignedDocument.should.have.property('nonce');
+    bobSignedDocument.should.have.property('proof');
+    bobSignedDocument.proof.should.have.property('verificationMethod');
+    // the document was ultimately signed with alice's key
+    bobSignedDocument.proof.verificationMethod.should.equal(aliceKey.id);
+
+    // Bob has successfully used alice's key to sign a document!
+
+    // Bob now delegates the use of Alice's key to Carol
+    const carolZcap = {
+      '@context': bedrock.config.constants.SECURITY_CONTEXT_V2_URL,
+      // this is a unique ID
+      id: `urn:zcap:${uuid()}`,
+      invoker: carolKey.id,
+      // the capability Alice gave to Bob
+      parentCapability: zcap.id,
+      // this is where we need to ensure the allowedAction here is included
+      // in the allowedAction of the parentCapability, there is an issue in
+      // ocapld for this.
+      allowedAction: 'sign',
+      invocationTarget: zcap.invocationTarget,
+    };
+
+    // finish bobs delegation to carol
+    const signedCapabilityFromBobToCarol = await _delegate({
+      capabilityChain: [
+        aliceKey.kmsId,
+        zcap,
+      ],
+      signer: bobKey,
+      zcap: carolZcap
+    });
+
+    // Bob uses bedrock-web-zcap-storage to store record of the
+    // delegation to Carol
+    await helpers.storeDelegation({
```

Comment:
this may need to be modified to store in EDV, or possibly removed from this test if it over-complicates things.